### PR TITLE
usage reporting: report utilized CPU time

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

This takes a similar approach to Loki's https://github.com/grafana/loki/pull/12970 and replaces number of CPU cores to the actual CPU utilization

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by 56quarters** [comment]
Why does usage reporting need to include CPU usage? This seems like operation data that isn't appropriate to include in usage stats and something better done with metrics.

**Comment by dimitarvdimitrov** [comment]
this is for GEM billing purposes. That's what Loki (and GEL) did, so we followed suit. I think it's equally viable to do this in GEM only.

**Comment by pr00se** [comment]
The usage report already contains:
- The Mimir memory utilization and number of goroutines.
- The number of logical CPU cores available to the Mimir process.

This is an improvement to the second item, bringing it more inline with the first (usage rather than raw available core count).

File: pkg/usagestats/report.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func buildMetrics() map[string]interface{} {
 		"num_cpu":       runtime.NumCPU(),
```
Comment (pr00se) [inline pkg/usagestats/report.go:84]: What do you think of removing this in lieu of the new stat?

Comment (dimitarvdimitrov) [reply pkg/usagestats/report.go:84]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@
   * Query blocking can no longer be circumvented with an equivalent query in a different format; see [Configure queries to block](https://grafana.com/docs/mimir/latest/configure/configure-blocked-queries/)
 * [CHANGE] Query-frontend: stop using `-validation.create-grace-period` to clamp how far into the future a query can span.
 * [CHANGE] Clamp [`GOMAXPROCS`](https://pkg.go.dev/runtime#GOMAXPROCS) to [`runtime.NumCPU`](https://pkg.go.dev/runtime#NumCPU). #8201
+* [CHANGE] Anonymous usage statistics tracking: add CPU usage percentage tracking. #8282
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:16]:
```suggestion
* [CHANGE] Anonymous usage statistics tracking: replace available CPU cores with CPU usage percentage. #8282
```

**Review by jdbaldry — APPROVED** [review]
Docs changes lgtm

**Review by pr00se — APPROVED** [review]
LGTM, thank you!

**Pull request merged**